website/src/_posts/2021-08-30.31.md

```diff
@@ -0,0 +1,250 @@
+
+// Invalid event
+uppy.on<Meta>('upload-errrOOOoooOOOOOrrrr', () => {
+  // ...
+})
+
+// After:
+
+// Normal event signature
+uppy.on('complete', (result) => {
+  const successResults = result.successful
+})
+
+// Custom signature
+type Meta = { myCustomMetadata: string }
+
+// Notice how the custom type has now become the second argument
+uppy.on<'complete', Meta>('complete', (result) => {
+  // The passed type is now merged into the `meta` types.
+  const meta = result.successful[0].meta.myCustomMetadata
+})
+```
+
+Using official Uppy plugins that add their own events are automatically added to the namespace. For instance, when using [`@uppy/dashboard`][dashboard]:
+
+<!-- eslint-disable @typescript-eslint/no-unused-vars -->
+```ts
+uppy.on('dashboard:file-edit-state', (file) => {
+  const fileName = file.name
+})
+```
+
+## Batch pre-signing URL's for AWS S3 Multipart
+
+The [`@uppy/aws-s3-multipart`][aws-s3-multipart] plugin can be used to upload files directly to an S3 bucket using S3’s Multipart upload strategy. With this strategy, files are chopped up in parts of 5MB+ each, so they can be uploaded concurrently. It is also very reliable: if a single part fails to upload, only that 5MB chunk has to be retried.
+
+However, every part needed to make a trip to the server to generate a pre-signed URL. This means a 1GB file uploaded in 5mb chunks would require 200 trips to the server.
+
+As of Uppy 2.0.0, we now pre-sign URLs in batches. That same 1GB file now only takes 50 trips to the server (if the limit was 4).
+
+This is now the new default. Thanks to [@martin-brennan](https://github.com/martin-brennan) for making this contribution.
+
+Care about reliable uploads? You could also consider [`@uppy/tus`][tus] with a self-hosted or Transloadit Tus server. Tus can resume uploads, support smaller chunks, and higher upload speeds.
+
+## And more
+
+- The `.run` method on the `Uppy` instance has been removed. This method was already obsolete and only logged a warning. As of this major version it no longer exists.
+
+- [`@uppy/informer`][informer] now supports showing multiple notifications at the same time. The notifications themselves have also been improved.
+
+- Improved screenreader accessibility for checkboxes and the remove file button for [`@uppy/dashboard`][dashboard].
+
+- Sort files and folders alphabetically in the Google Drive provider.
+
+- Polished our code base with improved eslint rules, private field methods, and other modern JavaScript features that help us write more intention revealing and safe code.
+
+- To make Uppy more friendly towards new contributors, we have renamed our `master` branch to `main`.
+
+## What future remains for 1.0
+
+1.0 will continue to receive bug fixes for 3 more months (until 1 December 2021), security fixes for 1 more year (until 1 September 2022), but, no new features as of today. Exceptions are unlikely but _can_ be made, for example to accomodate those with commercial support contracts.
```

Comment:
nit: ISO-8901 / RFC 3339 date format
```suggestion
1.0 will continue to receive bug fixes for 3 more months (until 2021-12-01), security fixes for 1 more year (until 2022-09-01), and no new features as of today. Exceptions are unlikely but _can_ be made, for example to accomodate those with commercial support contracts.
```

Reply:
Readability wise I don't think this is an improvement

Reply:
We should at least use [`<time>`](https://developer.mozilla.org/en-US/docs/Web/HTML/Element/time) then.
```suggestion
1.0 will continue to receive bug fixes for 3 more months (until <time datetime="2021-12-01">1 December 2021</time>), security fixes for 1 more year (until <time datetime="2022-09-01">1 September 2022</time>), and no new features as of today. Exceptions are unlikely but _can_ be made, for example to accomodate those with commercial support contracts.
```